### PR TITLE
Azure simple hw refactor

### PR DESCRIPTION
**Pull request by TechnicallyWilliams**
## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES/NO] I have updated the documentation accordingly.
* [YES/NO/NA] I have added tests to cover my changes.
* [YES/NO/NA] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES/] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------

Issue Number: #120 

More:
- app-service module main.tf:
  - This module currently creates a access restriction vnet_subnet_id by default. 
- hello world template:
  - The hello world template is not concerned with creating an access restriction vnet_subnet_id

## What is the new behavior?
-------------------------------------

Azure Simple Hello World template now deploys an Azure app-service plan and app service using existing Cobalt modules for both resources.
- hello world template output.tf
  - Output file now accesses index of expected app-service names map in order to generate an output.
  - Template now generates more resources due to default settings of app-service plan and app service modules.
- app-service module main.tf:
   - Hello World's "Enable storage option" was relocated away from the template into app-service module definition.
   - access restriction vnet_subnet_id creation for this module now keys off of rather or not vnet_name is provided.

## Does this introduce a breaking change?
-------------------------------------
- [NO]

## Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

## Other information
-------------------------------------

AZURE SIMPLE HELLO WORLD
image

AZURE SIMPLE
image

### CODE REVIEW

File: test-harness/infratests/integration.go [hunk 1]
```diff
@@ -50,8 +50,8 @@ func RunIntegrationTests(fixture *IntegrationTestFixture) {
 	defer terraform.RunTerraformCommand(fixture.GoTest, fixture.TfOptions, "workspace", "delete", workspaceName)
 	defer terraform.WorkspaceSelectOrNew(fixture.GoTest, fixture.TfOptions, "default")
 
-	terraform.Apply(fixture.GoTest, fixture.TfOptions)
 	defer terraform.Destroy(fixture.GoTest, fixture.TfOptions)
+	terraform.Apply(fixture.GoTest, fixture.TfOptions)
```
Comment (nmiodice) [inline test-harness/infratests/integration.go]: Can you rebase off master? This change has been merged

File: test-harness/infratests/unit.go [hunk 2]
```diff
@@ -136,7 +136,7 @@ func validatePlanResourceKeyValues(fixture *UnitTestFixture, plan *terraformCore
 			}
 		}
 	}
-
+	fmt.Println(seen)
```
Comment (nmiodice) [inline test-harness/infratests/unit.go]: We have a task to add this debugging (https://github.com/microsoft/orion/issues/21). I think we should focus the effort for that task as proper logging will be more than dumping the output here. My suggestion is to remove this line.

File: infra/modules/providers/azure/app-service/variables.tf [hunk 3]
```diff
@@ -20,6 +20,12 @@ variable "app_service_name" {
   default     = {}
 }
 
+variable "enable_storage" {
+  description = "Determines whether or not a storage is attached to the app service."
+  type        = "string"
+  default     = "true"
```
Comment (erikschlegel) [inline infra/modules/providers/azure/app-service/variables.tf]: This should default to `false`

File: .gitignore [hunk 4]
```diff
@@ -346,6 +346,7 @@ ASALocalRun/
 
 # VSCode Files
 .vscode/
+.vscode-insiders
```
[the remark this answers is not in the document]
Comment (TechnicallyWilliams) [reply .gitignore]: I followed your suggestion. Good ole' wildcards. Thanks!

File: infra/modules/providers/azure/app-service/main.tf [hunk 5]
```diff
@@ -51,7 +52,7 @@ resource "azurerm_app_service_slot" "appsvc_staging_slot" {
 
 resource "azurerm_template_deployment" "access_restriction" {
   name                = "access_restriction"
-  count               = "${length(keys(var.app_service_name))}"
+  count               = "${(length(var.subnet_names) >= "1") ? length(keys(var.app_service_name)) : 0}"
```
[the remark this answers is not in the document]
Comment (TechnicallyWilliams) [reply infra/modules/providers/azure/app-service/main.tf]: arm deployment block for configuring "access_restriction" now keys off of "vnet name" instead.

File: infra/modules/providers/azure/app-service/variables.tf [hunk 6]
```diff
@@ -62,8 +68,14 @@ variable "vnet_name" {
   default     = ""
 }
 
+variable "subnet_names" {
```
[the remark this answers is not in the document]
Comment (TechnicallyWilliams) [reply infra/modules/providers/azure/app-service/variables.tf]: Thanks! Per our discussion, "subnet_names" was removed and replaced with "vnet_name".

File: infra/templates/azure-simple-hw/tests/integration/azure_simple_test.go [hunk 7]
```diff
@@ -32,7 +33,7 @@ var tfOptions = &terraform.Options{
 // Validates that the service responds with HTTP 200 status code. A retry strategy
 // is used because it may take some time for the application to finish standing up.
 func httpGetRespondsWith200(goTest *testing.T, output infratests.TerraformOutput) {
-	hostname := output["app_service_default_hostname"].(string)
+	hostname := "https://" + output["app_service_default_hostname"].(string)
```
[the remark this answers is not in the document]
Comment (TechnicallyWilliams) [reply infra/templates/azure-simple-hw/tests/integration/azure_simple_test.go]: Hard reference to "https" was removed from the go script. Test now asserts "http" generated from 
azure simple hw output.tf.

https://github.com/microsoft/cobalt/pull/162/files#diff-277d2665c760f0faa7335cf338deef6a

File: test-harness/infratests/unit.go (hunk 2 above)
Comment (TechnicallyWilliams) [reply test-harness/infratests/unit.go]: removed

File: infra/modules/providers/azure/app-service/variables.tf (hunk 3 above)
Comment (TechnicallyWilliams) [reply infra/modules/providers/azure/app-service/variables.tf]: Latest now defaults to "false".

**Pull request merged**